runner/src/server/plugins/engine/plugin.ts

```diff
@@ -186,9 +187,63 @@ export const plugin = {
       },
     });
 
+    const queryParamPreHandler = async (
+      request: HapiRequest,
+      h: HapiResponseToolkit
+    ) => {
+      const { query } = request;
+      const { id } = request.params;
+      const model = forms[id];
+      if (!model) {
+        throw Boom.notFound("No form found for id");
+      }
+
+      const prePopFields = model.fieldsForPrePopulation;
+      if (
+        Object.keys(query).length > 0 &&
+        Object.keys(prePopFields).length > 0
+      ) {
+        const { cacheService } = request.services([]);
```

Comment:
use a guard statement. it looks like this if statement has a lot of nested ifs, guard statements will let you unnest more of the code so it's more readable. 

```
const shouldHandlePrePopulationFields = Object.keys(query).length > 0 && Object.keys(prePopFields).length > 0

if(!shouldHandlePrePopulationfields) {
  return h.continue
}
```